src/components/time-selector/TimeSelector.tsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react'
+import { checkboxArray } from './Checkboxes'
+import { ICheckbox, IDayCard } from './TimeSelectorInterfaces'
+import { TimeSelectorCard } from '../card/TimeSelectorCard'
+
+export const TimeSelector = () => {
+
+    const [cards, setCards] = useState<IDayCard[]>([])
+    const [checkboxes, setCheckboxes] = useState<ICheckbox[]>(checkboxArray)
+
+    const getCards = () => {
+        return ( 
+            cards?.map(card => {
+                const { name } = card
+                return <TimeSelectorCard key={name} name={name}/>
+            })
+        )
+    }
+
+    const markCheckbox = (checkboxValue: boolean, dayName: string, index: number) => {
+        checkboxes[index].checked = checkboxValue
+        setCheckboxes(checkboxes)
+    
+        if(checkboxValue) {
+            setCards([...cards, {name: dayName}])
+        } else {
+            for(let index in cards) {
```

Comment:
If we have sorted cards. I think we can avoid this loop.

Reply:
I don't think.  because we cannot remove card using checkbox index. and we are removing card on bases of checkboxes.

Reply:
Let's first sort the cards then we'll see whether we can avoid this loop or not.

Reply:
I've pushed code.